### PR TITLE
refactor exercises and style exercises from notes

### PR DESCRIPTION
**Pull request by wilkus27**
Issue: https://github.com/openstax/cnx-recipes/issues/5569

### CODE REVIEW

**Review by jbwilson8 — APPROVED** [review]

File: styles/books/nursing-internal/book.scss [hunk 1]
```diff
@@ -304,11 +286,17 @@ $ChapterIntroType: fullWidth;
...
 
 // EOC
 @include use('KeyTerms', 'carnival:::KeyTermsShape');
-@include use('EocExercisesModules','carnival:::ExercisesModuleShape');
+@include carnival.exercises('.os-review-questions-container', $number-width: 2%);
+@include carnival.exercises('.os-check-understanding-container', $number-width: 2%);
+@include carnival.exercises('.os-critical-thinking-container', $number-width: 2%);
+@include carnival.exercises('.os-competency-based-container', $number-width: 2%);
+@include carnival.exercises('.os-reflection-questions-container', $number-width: 2%);
+@include carnival.exercises('.os-what-nurses-do-container', $number-width: 2%);
```
Comment (kswanson33) [inline styles/books/nursing-internal/book.scss:299]: hopefully these selectors can be listed in one string, like: 

```scss
@include carnival.exercises(
  '.os-review-questions-container, .os-check-understanding-container, ...', 
  $number-width: 2%
)
```

File: styles/lib/carnival/features/_exercises.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+// Private mixins
+@mixin -number($width: 0) {
+    display: table-cell;
+    text-decoration: none;
+    color: carnival.$black;
+    font-weight: bold;
+    @if $width != 0 {
+        width: $width;
+    }
```
Comment (kswanson33) [inline styles/lib/carnival/features/_exercises.scss:41]: can be replaced with `width: $width` -- shouldn't need the if-statement

File: styles/lib/carnival/features/_exercises.scss [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+@mixin -has-first-element() {
+    display: table-cell;
+    vertical-align: bottom;
+    padding-top: 0.7rem;
+    width: 100%;
+}
+
+@mixin -problem-or-question($number-width: 0) {
```
Comment (kswanson33) [inline styles/lib/carnival/features/_exercises.scss:53]: nice!

File: styles/books/nursing-internal/book.scss [hunk 4]
```diff
@@ -304,11 +286,17 @@ $ChapterIntroType: fullWidth;
 @include use('PsychosocialConsiderationsNote', 'carnival:::IconNoteShape');
 
 // Exercise in Note
-@include use('UnfoldingCaseStudyExercise','carnival:::ExercisesModuleShape');
+@include carnival.exercises('.unfolding-casestudy');
+@include carnival.exercises('.electronic-hr');
```
Comment (kswanson33) [inline styles/books/nursing-internal/book.scss:290]: can be 

```scss
@include carnival.exercises('.unfolding-casestudy, .electronic-hr')
```

?

**Review by kswanson33 — CHANGES_REQUESTED** [review]
small comments, looks awesome. thank you!!!

File: styles/lib/carnival/features/_exercises.scss [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                        display: inline;
+                    }
+                }
+            }
+        }
+    }
+}
+
+// Private mixins
+@mixin -number($width: null) {
```
Comment (wilkus27) [inline styles/lib/carnival/features/_exercises.scss:34]: Had to change to `null`. Without `if` statment and with `0` value property was not optional. It was appearing in notes exercises as well.

**Comment by wilkus27** [comment]
@kswanson33 Thank you for all the suggestions! It's ready for re-review :wink:

File: styles/lib/carnival/features/_exercises.scss (hunk 5 above)
Comment (kswanson33) [reply styles/lib/carnival/features/_exercises.scss:34]: ohhh I see, I misunderstood what you were trying to do. good to know that `null` as a default argument makes a property optional! thanks for figuring that out!

**Review by kswanson33 — APPROVED** [review]

**Pull request merged**